src/solvers/dgsem_tree/dg_2d_parabolic.jl

```diff
@@ -62,162 +91,142 @@ function transform_variables!(u_transformed, u, mesh::TreeMesh{2},
   end
 end
 
-# note: the argument parabolic_scheme is not a DG type; it contains solver-specific
-# information such as an LDG penalty parameter.
-function calc_divergence!(du, u, t, flux_viscous,
-                          mesh::TreeMesh{2}, equations_parabolic,
-                          boundary_conditions_parabolic, dg::DG,
-                          parabolic_scheme, # not a `DG` type
-                          cache, cache_parabolic)
-  # Reset du
-  @trixi_timeit timer() "reset ∂u/∂t" begin
-    reset_du!(du, dg, cache)
-  end
-
-  # Calculate volume integral
-  @trixi_timeit timer() "volume integral" begin
-    @unpack derivative_dhat = dg.basis
-    @threaded for element in eachelement(dg, cache)
 
-      flux_viscous_x, flux_viscous_y = flux_viscous
+# This is the version used when calculating the divergence of the viscous fluxes
+function calc_volume_integral!(du, flux_viscous,
+                               mesh::TreeMesh{2},
+                               nonconservative_terms,
+                               equations_parabolic::AbstractEquationsParabolic,
+                               volume_integral::VolumeIntegralWeakForm,
```

Comment:
So if I understand this correctly, the advective terms also need to use `VolumeIntegralWeakForm` in order for one to use the parabolic terms, because this function get passed `dg.volume_integral` on line 43 above. Is that right?

The parabolic part will always uses the of `VolumeIntegralWeakForm` but the advective part would potentially need `VolumeIntegralFluxDifferencing` or `VolumeIntegralShockCapturingHG`, depending on the application.

This is something to keep in mind.

Reply:
Good point. No, I don't think this makes sense, and I have removed this restriction in 9510dc8c.

Reply:
Should we also remove `nonconservative_terms` from the signature?